File: app/src/main/java/com/example/macarus0/walkiewalkie/data/WalkPhoto.java
package com.example.macarus0.walkiewalkie.data;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;

@Entity
public class WalkPhoto implements PhotoItem{
    @Ignore
    public long getId() {
        return photoId;
    }

    @PrimaryKey(autoGenerate = true)
    long photoId;

    public String getName() {return null;}

    public String getPhotoUri() {
        return photoUri;
    }

    public void setPhotoUri(String photoUri) {
        this.photoUri = photoUri;
    }

    public long getWalkId() {
        return walkId;
    }

    public void setWalkId(long walkId) {
        this.walkId = walkId;
    }

    String photoUri;
    long walkId;

}

File: app/src/main/java/com/example/macarus0/walkiewalkie/data/WalkPhoto.kt
package com.example.macarus0.walkiewalkie.data

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
class WalkPhoto : PhotoItem {

    @PrimaryKey(autoGenerate = true)
    var photoId: Long = 0

    override var id: Long = 0
        get() = photoId

    override var photoUri: String = ""
    override var name: String = ""
    var walkId: Long = 0
}
